lib/rebaser-server/src/change_set_processor_task.rs

```diff
@@ -0,0 +1,420 @@
+        DalTransactions(#[from] dal::TransactionsError),
+        #[error("error publishing reply: {0}")]
+        PublishReply(#[source] si_data_nats::Error),
+        /// Failures related to rebasing/updating a snapshot or change set pointer.
+        #[error("rebase error: {0}")]
+        Rebase(#[from] RebaseError),
+        #[error("error serializing: {0}")]
+        Serialize(#[from] SerializeError),
+        /// When failing to find the workspace
+        #[error("workspace error: {0}")]
+        Workspace(#[from] dal::WorkspaceError),
+        /// When failing to do an operation using the [`WorkspaceSnapshot`]
+        #[error("workspace snapshot error: {0}")]
+        WorkspaceSnapshot(#[from] dal::WorkspaceSnapshotError),
+        /// When failing to send a [`WsEvent`]
+        #[error("failed to construct ws event: {0}")]
+        WsEvent(#[from] dal::WsEventError),
+    }
+
+    type Error = HandlerError;
+
+    type Result<T> = result::Result<T, HandlerError>;
+
+    impl IntoResponse for HandlerError {
+        fn into_response(self) -> Response {
+            // TODO(fnichol): there are different responses, esp. for expected interrupted
+            error!(si.error.message = ?self, "failed to process message");
+            Response::internal_server_error()
+        }
+    }
+
+    pub(crate) async fn default(
+        State(state): State<AppState>,
+        HeaderReply(maybe_reply): HeaderReply,
+        ApiTypesNegotiate(request): ApiTypesNegotiate<EnqueueUpdatesRequest>,
+    ) -> Result<()> {
+        let AppState {
+            workspace_id,
+            change_set_id,
+            nats,
+            ctx_builder,
+            run_notify,
+        } = state;
+        let mut ctx = ctx_builder
+            .build_for_change_set_as_system(workspace_id.into(), change_set_id.into())
+            .await?;
+
+        let rebase_status = perform_rebase(&mut ctx, &request)
+            .await
+            .unwrap_or_else(|err| {
+                error!(
+                    si.error.message = ?err,
+                    ?request,
+                    "performing rebase failed, attempting to reply",
+                );
+                RebaseStatus::Error {
+                    message: err.to_string(),
+                }
+            });
+
+        // Dispatch eligible actions if the change set is the default for the workspace.
+        // Actions are **ONLY** ever dispatched from the default change set for a workspace.
+        if matches!(rebase_status, RebaseStatus::Success { .. }) {
+            // If we find dependent value roots, then notify the serial dvu task to run at least
+            // one more dvu
+            if ctx
+                .workspace_snapshot()?
+                .has_dependent_value_roots()
+                .await?
+            {
+                run_notify.notify_one();
+            }
+
+            if let Some(workspace) = Workspace::get_by_pk(&ctx, &workspace_id.into()).await? {
```

Comment:
I feel like we should fail if this is empty, unless you're accounting for workspace "deletion". Even then, I suspect it would be a soft delete and this should always return `Some`.

Reply:
Yeah I wonder--this code was mostly lifted off prior versions of the Rebaser so I didn't question what it was doing all that much. I wonder what @zacharyhamm or @jhelwig might think?